TACC/Core-CMS
# Task/FP-1289 Image Grid Pattern

### Overview

Create class for CMS editors to use to create image grids (but _not_ galleries).

### Issue

[FP-1289](https://jira.tacc.utexas.edu/browse/FP-1289)

### Changes

- New `o-grid` CSS class (and sample markup).
- New snippet (symlink to CSS sample markup).
- New global variable for grid gap (match Bootstrap value).
- New `x-grid` CSS mixin.

### Screenshots

| All Snippets | Example Gallery |
| - | - |
|  ![FP-1289 Four Snippets](https://user-images.githubusercontent.com/62723358/144911465-4f3ea433-fb24-45c1-b04b-2650f08cc4d9.jpeg) | ![FP-1289 Example - Center, Auto](https://user-images.githubusercontent.com/62723358/144911470-14d463c5-cb34-455d-b60e-c2763e09acba.png) |

### Testing

__Remote:__
<details><summary>Jenkins Jobs</summary>

- Build: https://jenkins01.tacc.utexas.edu/job/Core_CMS/322
- Deploy: https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/939/

</details>

1. Open https://dev.cep.tacc.utexas.edu/design-system/pattern-library/o-grid/.
2. If this branch is deployed, then the page should load many sets of collapsed galleries.
3. Follow "Remote & Local" steps.

__Local:__
<details><summary>Snippets to Make</summary>

> __NAME: `UI: O-Grid: fill, auto`__
> HTML: `o-grid--image-fill o-grid--col-auto-count`
> TEMPLATE: `snippets/manual-pattern-library/o-grid.html`

> __NAME: `UI: O-Grid: center, auto`__
> HTML: `o-grid--center-align o-grid--col-auto-count`
> TEMPLATE: `snippets/manual-pattern-library/o-grid.html`

> __NAME: `UI: O-Grid: fill, auto`__
> HTML: `o-grid--image-fill o-grid--col-auto-count`
> TEMPLATE: `snippets/manual-pattern-library/o-grid.html`

> __NAME: `UI: O-Grid: fill, min`__
> HTML: `o-grid--center-align o-grid--col-min-width`
> TEMPLATE: `snippets/manual-pattern-library/o-grid.html`

</details>

1. Be familiar with how to [Test CMS Changes](https://github.com/TACC/Core-CMS/wiki/Test-CMS-Changes).
2. Create snippets that uses template `snippets/manual-pattern-library/o-grid.html`.
3. Follow "Remote & Local" steps.

__Remote & Local:__

1. Expand each gallery within each section/snippet.
2. Confirm that each class combination[^1] does what [each class says it should](https://github.com/TACC/Core-CMS/blob/task/FP-1289-image-grid-gallery/taccsite_cms/static/site_cms/css/src/_imports/objects/o-grid.css#L9-L12).
    - To test each class individually, you can reduce the classnames passed in the HTML snippets.[^2]
    - To test number of columns, resize window. For `o-grid--col-auto-count`, test [standard breakpoints](https://confluence.tacc.utexas.edu/x/b4AZCg).

[^1]: Most uses of `o-grid` will likely have two of these modifier classes.
[^2]: ~~To preview this, see [tacc-wbomar-css-pattern-library-kss.surge.sh](http://tacc-wbomar-css-pattern-library-kss.surge.sh/section-objects.html#kssref-objects-grid).~~ (This site is currently being used to preview `c-recognition`.)
